From a0e40120f06cadf97737789a72106ce8017d3a01 Mon Sep 17 00:00:00 2001
From: Andy Shaw <andy.shaw@qt.io>
Date: Thu, 7 Mar 2024 08:27:24 +0100
Subject: [PATCH] XCB: Unset the connection's mousePressWindow if it's this
 window

In some rare cases it can delete the window before this is handled by
other calls, so rather than crashing we clean up.

Change-Id: Iddcd9890f0c13f4130626b0ed9c5b32f5890208d
Reviewed-by: Axel Spoerl <axel.spoerl@qt.io>
(cherry picked from commit b8f9a8681347476f4db1719b38a67d7f9d3c74be)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
(cherry picked from commit be78514b206aff3152a239c188fcc4e7aba5439a)
(cherry picked from commit a86cd4dbeeacc42f095db5ada0f0ea59f390465c)
(cherry picked from commit a6ff25079d5aeb6655f4c301c52d90304d1b3745)
(cherry picked from commit 3707fbdbc9f38ea91f0dfedce1ad7c4f8d27d0ad)
---
 src/plugins/platforms/xcb-maemo/qxcbwindow.cpp | 2 ++
 1 file changed, 2 insertions(+)

Index: qt-platform-maemo/src/plugins/platforms/xcb-maemo/qxcbwindow.cpp
===================================================================
--- qt-platform-maemo.orig/src/plugins/platforms/xcb-maemo/qxcbwindow.cpp
+++ qt-platform-maemo/src/plugins/platforms/xcb-maemo/qxcbwindow.cpp
@@ -556,6 +556,8 @@ void QXcbWindow::destroy()
         doFocusOut();
     if (connection()->mouseGrabber() == this)
         connection()->setMouseGrabber(nullptr);
+    if (connection()->mousePressWindow() == this)
+        connection()->setMousePressWindow(nullptr);
 
     if (m_syncCounter && connection()->hasXSync())
         xcb_sync_destroy_counter(xcb_connection(), m_syncCounter);
